torch/_dynamo/symbolic_convert.py

```diff
@@ -1086,10 +1086,23 @@ def run(self):
             except TensorifyScalarRestartAnalysis:
                 raise
             except BackendCompilerFailed:
+                raise
+            except RuntimeError as e:
+                if hasattr(e, "msg") and "Could not guard on data-dependent" in e.msg:
+                    print(
+                        "\n"
+                        + self.output.graph.python_code(
```

Comment:
there's also `self.output.print_readable(print_output=False)`. We might want to use that instead? Mainly because:

(1) it'll be more consistent with the sort of printed graph outputs we see in TORCH_LOGS and tlparse (to be fair I'm not sure if/what subtle differences show up when you print the python code)

(2) it's exercised more in the PT2 codepaths, so there's probably a bit less risk of eg. `print_readable()` choking on a weird graph and raising an error (to be fair I'm not sure how real of a risk this is with `python_code()`)

Reply:
I assume you mean `print_readable` on graphmodule? I originally did that but, at least for the non strict tracing use case, CI would fail on some tests because there are implicit dependencies on the passed in `root` so you can't just fabricate one with {} or nn.Module(). Maybe dynamo has stricter guarantees? I just updated the PR accordingly.

Reply:
sg - at the very least it looks like we have precedent for this in dynamo, e.g. here: https://github.com/pytorch/pytorch/blob/main/torch/_dynamo/eval_frame.py#L1655

One example of where this can be bad is if your graph has `call_module()` calls in it - that will break because the `nn.Module()` that you pass in needs to have the same submodules in it. Dynamo should already give us the guarantee that its graph is flattened and does not have submodules though